Laboratory work #7, Sviridov Vyacheslav - 20FPL1

VSvir opened the pull request.

VSvir committed:
Update main.py

VSvir committed:
fill settings file

VSvir committed:
[feature] implement dataset loading

VSvir committed:
[feature] implement dataset analysis
[refactor] minor changes in data importer implementation

VSvir committed:
[feature] implement dataset analysis pipeline
[refactor] change dataset name in settings file
[refactor] optimize imports

SalnikovIgor committed:
Merge branch 'fipl-hse:main' into main

VSvir committed:
[feature] implement dataset preprocessing steps

VSvir committed:
Merge remote-tracking branch 'origin/main'

SalnikovIgor committed:
Merge branch 'fipl-hse:main' into main

SalnikovIgor commented:
Fix unittests

VSvir committed:
Merge remote-tracking branch 'origin/main'

SalnikovIgor committed:
Merge branch 'fipl-hse:main' into main

SalnikovIgor committed:
Merge branch 'fipl-hse:main' into main

SalnikovIgor commented:
Time to start the task at 6

VSvir committed:
Merge remote-tracking branch 'origin/main'

VSvir committed:
[feature] implement TaskDataset class

VSvir committed:
[fix] save preprocessed dataset to _data attribute of Preprocessor class

VSvir committed:
[feature] implement model analysis

VSvir committed:
[chore] update requirements
[feature] add model analysis steps in start.py

VSvir commented:
So, here's what I have for now. I don't understand where I should look for `max_context_length` property though and my PC feels bad while trying to get model summary on 64 batches as defined in tests files. Guess I'll discuss these issues with others tomorrow

VSvir commented:
Oh, btw, should we check `output_shape` property by taking the size of the output layer?

VSvir committed:
[refactor] sort dependencies in requirements.txt

SalnikovIgor commented on lab_7_llm/settings.json:
```diff
@@ -1,8 +1,8 @@
+    "metrics": ["accuracy"]
   },
-  "target_score": 0
+  "target_score": 4
```
I think it should change to 6?

VSvir replied:
Yes, I'll change it when I'll complete all tasks for this mark

SalnikovIgor reviewed (changes requested):
Implement inference sample

SalnikovIgor commented:
> So, here's what I have for now. I don't understand where I should look for `max_context_length` property though and my PC feels bad while trying to get model summary on 64 batches as defined in tests files. Guess I'll discuss these issues with others tomorrow

We take the `max_context_length` parameter from the model config. As for the inference with batch 64, I think if you have experiments with a smaller batch, then you should pass tests with a batch of 64

SalnikovIgor commented:
> Oh, btw, should we check `output_shape` property by taking the size of the output layer?

I don't quite understand the question, we can discuss it at the seminar

VSvir commented:
Thanks for replies, I've also asked my questions in chat, so I believe we'll figure everything out

SalnikovIgor committed:
Merge branch 'fipl-hse:main' into main

VSvir committed:
[refactor] change part of the model analysis logic

VSvir committed:
[refactor] change descriptions of custom functions

VSvir committed:
[feature] implement infer_sample function logic

SalnikovIgor committed:
Merge branch 'fipl-hse:main' into main

VSvir committed:
[refactor] remove printing of model summary
[feature] implement sample inference logic in start.py
[refactor] change target score

VSvir committed:
Merge remote-tracking branch 'origin/main'

VSvir committed:
[refactor] optimize imports

VSvir committed:
[feature] implement infer_batch function
[feature] implement inference on dataset in start.py

VSvir committed:
[refactor] shorten line reaching over limit

VSvir committed:
[feature] implement infer_batch function
[refactor] change inference functions logic to use infer_batch function

SalnikovIgor commented on lab_7_llm/start.py:
```diff
@@ -2,15 +2,43 @@
+
+    sample = dataset[randint(0, len(dataset))]
+    sample_inference = pipeline.infer_sample(sample)
+    print(f'SAMPLE: {sample}\nPREDICTION: {sample_inference}')
```
This is information for debugging?

VSvir replied:
Well, yes, but actually no
While I do use these prints for debugging, I was also planning to leave them here just to have a possibility to see the results of the previous steps which are required for lesser marks

VSvir replied:
Actually, this is the purpose of all prints in `start.py`

SalnikovIgor reviewed (changes requested):
Delete the debug prints, wait for the references to be updated and fix the tests

SalnikovIgor committed:
Merge branch 'fipl-hse:main' into main

VSvir committed:
[feature] implement evaluation logic
[refactor] update custom functions documentation
[feature] implement evaluation logic in start.py
[chore] update requirements.txt
[chore] update target score

VSvir committed:
[refactor] optimize imports

VSvir committed:
[refactor] optimize imports in start.py

VSvir committed:
[refactor] optimize imports in start.py again

VSvir committed:
[refactor] update main.py to fix mypy checks

VSvir committed:
[refactor] update data importer

SalnikovIgor committed:
Merge branch 'fipl-hse:main' into main

VSvir committed:
[refactor] remove pd.concat from infer_dataset function
[refactor] get max_context_length from max_length
[refactor] change batch size in start.py

VSvir committed:
[refactor] roll back unintended changes in model analysis test

SalnikovIgor committed:
Merge branch 'fipl-hse:main' into main

VSvir committed:
[refactor] change input shape order in model analysis
IDK what's wrong with this test, everything is OK locally

VSvir committed:
[refactor] fix model analysis
welp...

VSvir commented:
🎉

SalnikovIgor commented on requirements.txt:
```diff
@@ -0,0 +1,13 @@
+scikit-learn==1.4.0
+torch==2.1.2
+torchinfo==1.8.0
+tqdm==4.66.1
```
Also, why do we need `tqdm`?

SalnikovIgor commented on requirements.txt:
```diff
@@ -0,0 +1,13 @@
+evaluate==0.4.1
+numpy==1.26.3
+pandas==2.1.4
+pydantic==2.5.3
```
Why do we need `pydantic`?

SalnikovIgor reviewed (changes requested):
Fix minor comments and when do you plan to implement a service for a score of 10?

SalnikovIgor commented:
Also don't forget to remove debbug prints

VSvir committed:
[chore] remove unnecessary requirements
[refactor] remove prints from start.py due to comments

VSvir commented:
Oh, looks like we've broken the website

VSvir commented:
> when do you plan to implement a service for a score of 10?

![image](https://github.com/fipl-hse/2023-hello-llm/assets/90447390/53a4c5ea-8dc6-4eee-b08a-d34146e99e2c)

VSvir committed:
[refactor] update batch size
Let's see if the check will pass

VSvir committed:
[refactor] minor change to try again
IDK what's wrong

VSvir commented:
Bruh 🗿

VSvir committed:
[feature] initialize service core
[chore] add files for service implementation
[chore] update requirements.txt

VSvir committed:
[feature] implement service logic
[chore] update requirements.txt
[refactor] fix possible issue in start.py
[refactor] improve LLMPipeline logic

VSvir committed:
[refactor] pass empty DataFrame to pipeline in service.py

VSvir committed:
[refactor] update service.py to pass tests

VSvir committed:
[feature] implement main.js

VSvir committed:
[feature] implement web service frontend
[refactor] update paths
[refactor] update labels mapping
[refactor] minor changes in JS
[chore] update target score

VSvir committed:
[chore] update requirements.txt

VSvir committed:
[chore] add docstring to function in main.py
[refactor] minor changes in service.py

VSvir committed:
[refactor] yeah, imports

VSvir commented:
🎉🎉🎉

VSvir committed:
[refactor] update styles

VSvir committed:
[refactor] update service logic to get rid of bypass for checks

VSvir commented:
That's it 👍🏻

VSvir committed:
[refactor] remove unnecessary print

VSvir commented:
Now that's actually it

SalnikovIgor commented on lab_7_llm/assets/main.js:
```diff
@@ -0,0 +1,28 @@
+            question: premise.value,
+            hypothesis: hypothesis.value
+        })
+    }).then(res => {return res.json()})
```
I think this is the third time I've seen this code.
`async \ await` function was invented as the development of promises and getting rid of `then` and promise hell

SalnikovIgor commented on lab_7_llm/main.py:
```diff
@@ -32,6 +27,7 @@ class RawDataImporter(AbstractRawDataImporter):
     """
     A class that imports the HuggingFace dataset.
     """
+    _raw_data: DataFrame
```
This parameter is already created in the class `AbstractRawDataImporter`

VSvir replied:
Yes, but without overriding a type of this parameter Mypy checks were failing

SalnikovIgor commented on lab_7_llm/start.py:
```diff
@@ -2,15 +2,49 @@
+
+    evaluator = TaskEvaluator(
+        data_path,
+        [Metrics[metric.upper()] for metric in configs['parameters']['metrics']]
```
Do you think can we  pass this parameter differently?

VSvir replied:
I guess so, probably we could pass a whole Enum here and then look for a needed metric in `TaskEvaluator`, but this approach needs us to open a settings file in `main.py` module which I would prefer to refrain from

VSvir committed:
[refactor] improve JS code
